math/src/main/kotlin/ktx/math/ImmutableVector2.kt

```diff
@@ -0,0 +1,182 @@
+@file:Suppress("NOTHING_TO_INLINE")
+
+package ktx.math
+
+import com.badlogic.gdx.math.MathUtils
+import com.badlogic.gdx.math.Matrix3
+import com.badlogic.gdx.math.Vector2
+import kotlin.math.*
+
+/**
+ * Represent an **immutable** vector
+ *
+ * @property x the x-component of this vector
+ * @property y the y-component of this vector
+ */
+data class ImmutableVector2(val x: Float, val y: Float) : ImmutableVector<ImmutableVector2> {
+
+    override val len2: Float = Vector2.len2(x, y)
+
+    override fun isZero(margin: Float): Boolean = (x == 0f && y == 0f) || len2 < margin
+
+    override operator fun minus(other: ImmutableVector2): ImmutableVector2 = minus(other.x, other.y)
+
+    /** Returns the result of subtracting the given vector from this vector */
+    fun minus(deltaX: Float = 0f, deltaY: Float = 0f): ImmutableVector2 = ImmutableVector2(x - deltaX, y - deltaY)
+
+    override operator fun plus(other: ImmutableVector2): ImmutableVector2 = plus(other.x, other.y)
+
+    /** Returns the result of adding the given vector from this vector */
+    fun plus(deltaX: Float = 0f, deltaY: Float = 0f): ImmutableVector2 = ImmutableVector2(x + deltaX, y + deltaY)
+
+    override fun inc(): ImmutableVector2 = copy(x = x + 1, y = y + 1)
+    override fun dec(): ImmutableVector2 = copy(x = x - 1, y = y - 1)
+
+    override operator fun times(scalar: Float): ImmutableVector2 = times(scalar, scalar)
+    override operator fun times(vector: ImmutableVector2): ImmutableVector2 = times(vector.x, vector.y)
+
+    /** Returns this vector scaled by the given [xf] and [yf] factors */
+    fun times(xf: Float, yf: Float): ImmutableVector2 = ImmutableVector2(x * xf, y * yf)
```

Comment:
I understand that these were the method parameters in the original LibGDX code, but I'm not a fan of such non-obvious names. You could use `x`, `factorX` or `xFactor` instead.

Reply:
>I'm not a fan of such non-obvious names

Me neither. Actually in LibDGX they were named `x` and `y`. But i tend to find it even less obvious.

I'll go for `factorX`. (same for the method bellow)